st2common/st2common/util/db.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Extreme Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import mongoengine
+import six
+
+
+def mongodb_to_python_types(value):
+    if isinstance(value, mongoengine.base.datastructures.BaseDict):
+        value = dict(value)
+
+    if isinstance(value, mongoengine.base.datastructures.BaseList):
+        value = list(value)
+
+    if isinstance(value, dict):
+        value = {k: mongodb_to_python_types(v) for k, v in six.iteritems(value)}
+
+    if isinstance(value, list):
+        value = [mongodb_to_python_types(v) for v in value]
+
+    return value
```

Comment:
`value` passed is only one type of instance. Can it be checked with `if..else`.

Reply:
I'm not sure what you mean.  `value` can be any number of types.  The function recursively converts any BaseDict to dict and BaseList to list and if it's a dict/list then drill down to the values and convert those as well.  Anything other type, it will just leave the value as is.

Reply:
```
if isinstance(value, mongoengine.base.datastructures.BaseDict):
        value = dict(value)
elif isinstance(value, mongoengine.base.datastructures.BaseList):
        value = list(value)
elif isinstance(value, dict):
        value = {k: mongodb_to_python_types(v) for k, v in six.iteritems(value)}
else isinstance(value, list):
        value = [mongodb_to_python_types(v) for v in value]
```

Reply:
Oh I see what you mean. Yes, I will change.

Reply:
I still have to split it to two if-else block.  For example, after the convert from BaseDict to become dict, it needs to be traversed.  

```
if isinstance(value, mongoengine.base.datastructures.BaseDict):
        value = dict(value)
elif isinstance(value, mongoengine.base.datastructures.BaseList):
        value = list(value)

if isinstance(value, dict):
        value = {k: mongodb_to_python_types(v) for k, v in six.iteritems(value)}
else isinstance(value, list):
        value = [mongodb_to_python_types(v) for v in value]
```